orangecontrib/text/widgets/owscraper.py

```diff
@@ -0,0 +1,160 @@
+
+
+class OWScraper(OWWidget):
+
+
+	# Basic widget info
+    name = "Article Scraper"
+    description = "gets article, title, author, date etc. from the URL"
+    icon = "icons/Scraper.svg"
+    priority = 100
+    category="text"
+
+    # Input/output
+    inputs = [("Article_Links", Table, "set_data")]
+    outputs = [("Data", Table)]
+  
+    want_main_area = False
+   
+    # Settings
+    
+    # Output includes checkboxes.
+    includes_article = Setting(True)
+    includes_title = Setting(False)
+    includes_author = Setting(False)
+    includes_date = Setting(False)
+    includes_location = Setting(False)
+    includes_web_url = Setting(False)
+
+
+    def __init__(self):
+        super().__init__()
+        # To hold all the controls. Makes access easier.
+
+        self.scraper_controls = []
+        # Text includes box.
+        self.text_includes_box = gui.widgetBox(self.controlArea,
+                                               "Output", addSpace=True)
+        self.article_chbox = gui.checkBox(self.text_includes_box, self,
+                                           "includes_article", "Article")
+        self.author_chbox = gui.checkBox(self.text_includes_box, self,
+                                                 "includes_author",
+                                                 "Author")
+        self.date_chbox = gui.checkBox(self.text_includes_box, self,
+                                          "includes_date", "Date")
+        self.title_chbox = gui.checkBox(self.text_includes_box, self,
+                                           "includes_title", "Title")
+        self.web_url_chbox = gui.checkBox(self.text_includes_box, self,
+                                          "includes_web_url", "URL")
+
+        self.scraper_controls.append(self.article_chbox)
+        self.scraper_controls.append(self.author_chbox)
+        self.scraper_controls.append(self.date_chbox)
+        self.scraper_controls.append(self.title_chbox)
+        self.scraper_controls.append(self.web_url_chbox)
+
+
+        self.run_query_button = gui.button(self.controlArea, self, 'Run Query', tooltip="Run the chosen article Query")
+        self.run_query_button.setFocusPolicy(QtCore.Qt.NoFocus)
+        self.scraper_controls.append(self.run_query_button)
+
+
+    def date_to_str(input_date):
+        """
+        Returns a string representation of the input 
+        :param input_date:
+        :type input_date: datetime
+        :return: str
+        """
+        input_date = str(input_date.year) + str(input_date.month) + str(input_date.day) 
+        if input_date:
+            return input_date
+        else:
+            return"_"
+
+    def get_info(self, url):
+        """
+        Gets relevent available information from the url
+        :param url: 
+        :type url: string
+        :return : list
+        """
+        url_to_clean=url[4:-1] # remove irrevelent variables
```

Comment:
A better way to do it is to let the user decide in which columns are the URLs. For example, look at the GeoMap widget. There, the `Region Attribute` drop down menu determines in which column the countries are stored. This is far more general approach than fixing where they should be. Here we would need something similar. Once the feature is selected, you should access the URLs from the numpy array `Table.metas` rather than parsing the string representations.